addons/voice/org.openhab.voice.gtts/OSGI-INF/GoogleTTSService.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+	Copyright (c) 2010-2018 by the respective copyright holders.
+
+	All rights reserved. This program and the accompanying materials
+	are made available under the terms of the Eclipse Public License v1.0
+	which accompanies this distribution, and is available at
+	http://www.eclipse.org/legal/epl-v10.html
+
+-->
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" configuration-policy="optional" modified="updateConfig" immediate="true" name="org.openhab.gtts">
+	<service>
+		<provide interface="org.eclipse.smarthome.core.voice.TTSService"/>
+        <provide interface="org.eclipse.smarthome.config.core.ConfigOptionProvider"/>
+	</service>
+    <implementation class="org.openhab.voice.gtts.internal.GoogleTTSService"/>
+    <property name="service.pid" type="String" value="org.openhab.gtts"/>
+	<property name="service.config.description.uri" type="String" value="voice:gtts"/>
+	<property name="service.config.label" type="String" value="Google Cloud TTS Service"/>
+	<property name="service.config.category" type="String" value="voice"/>
```

Comment:
Please use DS annotations to generate these files.

E.g. add the following annotation on the GoogleTTSService class.

```java
@Component(immediate = true, service = { TTSService.class, ConfigOptionProvider.class }, property = {
        Constants.SERVICE_PID + "=org.openhab.gtts",
        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=voice:gtts",
        ConfigurableService.SERVICE_PROPERTY_LABEL + "=Google Cloud TTS Service",
        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=voice" })
```

Also put the `@Modified` annotation on the `GoogleTTSService.updateConfig` method.

Eclipse will then generate a `org.openhab.voice.gtts.internal.GoogleTTSService.xml` file in OSGI-INF with almost the same content.

Then delete this `GoogleTTSService.xml` file from the repository.

And add a .gitignore file to OSG-INF which ignores `*.xml` files. 

There is a Maven plugin which will still add the `org.openhab.voice.gtts.internal.GoogleTTSService.xml` to the bundle JAR. So we don't need to store it in the repository. :-)

Reply:
Can you test if just using `@Component` also works? 
* It looks the properties are nowadays part of the `TTSService` interface.
* Immediate doesn't need to be true because I think it will be activated when needed. 
* All interfaces that the class directly implements should also be provided so I think the `service` annotation property is also unnecessary.

The MacTTSService for instance doesn't need all these additional annotation properties, see

https://github.com/eclipse/smarthome/blob/de2f20106d9010f9c738c2c3fff07d70f8b8835f/extensions/voice/org.eclipse.smarthome.voice.mactts/src/main/java/org/eclipse/smarthome/voice/mactts/internal/MacTTSService.java#L42

Reply:
I've created https://github.com/openhab/openhab2-addons/pull/3723 to add DS annotations to the existing voice addons. :-)

Reply:
maybe the immediate is not needed (Ill check it later), but I think that the ConfigOptionProvider interface without the service annotation will not work...

Reply:
You were right. For annotation:
@Component(property = {
        Constants.SERVICE_PID + "=org.openhab.gtts",
        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=voice:gtts",
        ConfigurableService.SERVICE_PROPERTY_LABEL + "=Google Cloud TTS Service",
        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=voice" })

the build has generated

<?xml version="1.0" encoding="UTF-8"?>
<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.openhab.voice.gtts.internal.GoogleTTSService" modified="updateConfig">
    <property name="service.pid" value="org.openhab.gtts"/>
    <property name="service.config.description.uri" value="voice:gtts"/>
    <property name="service.config.label" value="Google Cloud TTS Service"/>
    <property name="service.config.category" value="voice"/>
    <service servicefactory="false">
        <provide interface="org.eclipse.smarthome.core.voice.TTSService"/>
        <provide interface="org.eclipse.smarthome.config.core.ConfigOptionProvider"/>
    </service>
    <implementation class="org.openhab.voice.gtts.internal.GoogleTTSService"/>
</scr:component>

It included even the ConfigOptionProvider...